osu.Server.BeatmapSubmission/BeatmapSubmissionController.cs

```diff
@@ -0,0 +1,273 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using MySqlConnector;
+using osu.Framework.Extensions;
+using osu.Game.IO.Archives;
+using osu.Server.BeatmapSubmission.Authentication;
+using osu.Server.BeatmapSubmission.Models.API.Requests;
+using osu.Server.BeatmapSubmission.Models.API.Responses;
+using osu.Server.BeatmapSubmission.Models.Database;
+using osu.Server.BeatmapSubmission.Services;
+using osu.Server.QueueProcessor;
+
+namespace osu.Server.BeatmapSubmission
+{
+    [Authorize]
+    public class BeatmapSubmissionController : Controller
+    {
+        private readonly IBeatmapStorage beatmapStorage;
+        private readonly BeatmapPackagePatcher patcher;
+
+        public BeatmapSubmissionController(
+            IBeatmapStorage beatmapStorage,
+            BeatmapPackagePatcher patcher)
```

Comment:
would still prefer this on one line as well

Reply:
Will apply for now. I will say that this is a tendency I've acquired from previous ASP.NET work - the more heavily you lean on DI in ASP.NET, which is done via ctor, the longer these argument lists get and at some point it gets way more sane to put every single one on its own line rather than do fifty params in three or four lines or whatever. But for now it's not a big deal.

Reply:
Yeah I can see it being fine when there's a lot of arguments, but I think I generally prefer single line until we hit a "this line is too long" "soft-wrap" we usually do.